static/app/components/contextPickerModal.tsx

```diff
@@ -411,102 +412,88 @@ type ContainerProps = SharedProps & {
    * List of slugs we want to be able to choose from
    */
   projectSlugs?: string[];
-} & DeprecatedAsyncComponent['props'];
-
-type ContainerState = {
-  organizations: Organization[];
-  integrationConfigs?: Integration[];
-  selectedOrganization?: string;
-} & DeprecatedAsyncComponent['state'];
-
-class ContextPickerModalContainer extends DeprecatedAsyncComponent<
-  ContainerProps,
-  ContainerState
-> {
-  getDefaultState() {
-    const storeState = OrganizationStore.get();
-    return {
-      ...super.getDefaultState(),
-      organizations: OrganizationsStore.getAll(),
-      selectedOrganization: storeState.organization?.slug,
-    };
-  }
+};
 
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {configUrl} = this.props;
-    if (configUrl) {
-      return [['integrationConfigs', configUrl]];
-    }
-    return [];
-  }
+export default function ContextPickerModalContainer(props: ContainerProps) {
+  const {allowAllProjectsSelection, configUrl, projectSlugs} = props;
 
-  componentWillUnmount() {
-    this.unlistener?.();
-  }
-
-  unlistener = OrganizationsStore.listen(
-    (organizations: Organization[]) => this.setState({organizations}),
-    undefined
+  const [organizations, setOrganizations] = useState<Organization[]>(
+    OrganizationsStore.getAll()
+  );
+  useEffect(() => {
+    const unsubscribe = OrganizationsStore.listen(setOrganizations, undefined);
+    return () => unsubscribe();
+  }, []);
+
+  const selectedOrganization = OrganizationStore.get();
+  const [selectedOrgSlug, setSelectedOrgSlug] = useState(
+    selectedOrganization.organization?.slug
   );
 
-  handleSelectOrganization = (organizationSlug: string) => {
-    this.setState({selectedOrganization: organizationSlug});
-  };
+  const {data, isError, isLoading, refetch} = useApiQuery<Integration[]>(
+    [configUrl ?? ''],
+    {
+      enabled: Boolean(configUrl),
+      staleTime: Infinity,
+    }
+  );
 
-  renderModal({
-    projects,
-    initiallyLoaded,
-    integrationConfigs,
-  }: {
-    initiallyLoaded?: boolean;
-    integrationConfigs?: Integration[];
-    projects?: Project[];
-  }) {
+  if (configUrl && isLoading) {
+    return <LoadingIndicator />;
+  }
+  if (configUrl && isError) {
+    return <LoadingError onRetry={refetch} />;
+  }
+  if (data?.length) {
```

Comment:
i think this is because you used isLoading
```suggestion
  if (data.length) {
```

Reply:
oh i guess because of enabled. hmmmm

Reply:
i think the checkes of `if (configUrl && isPending)` are messing with TS narrowing. I can probably shove that whole configUrl case into a sub component so the `enabled` flag isn't in the way as much